balancer/xds/lrs/lrs_test.go

```diff
@@ -0,0 +1,240 @@
+			}, {
+				dropCategories[0]: 59,
+				dropCategories[1]: 26,
+			}},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ls := newStore(testService)
+
+			for _, ds := range tt.drops {
+				var (
+					totalDropped uint64
+					droppedReqs  []*loadreportpb.ClusterStats_DroppedRequests
+				)
+				for cat, count := range ds {
+					totalDropped += count
+					droppedReqs = append(droppedReqs, &loadreportpb.ClusterStats_DroppedRequests{
+						Category:     cat,
+						DroppedCount: count,
+					})
+				}
+				want := []*loadreportpb.ClusterStats{
+					{
+						ClusterName:          testService,
+						TotalDroppedRequests: totalDropped,
+						DroppedRequests:      droppedReqs,
+					},
+				}
+
+				var wg sync.WaitGroup
+				for c, count := range ds {
+					for i := 0; i < int(count); i++ {
+						wg.Add(1)
+						go func(i int, c string) {
+							ls.callDropped(c)
+							wg.Done()
+						}(i, c)
+					}
+				}
+				wg.Wait()
+
+				if got := ls.buildStats(); !equalClusterStats(got, want) {
+					t.Errorf("lrsStore.buildStats() = %v, want %v", got, want)
+					t.Errorf("%s", cmp.Diff(got, want))
+				}
+			}
+		})
+	}
+}
+
+type lrsServer struct {
+	mu                sync.Mutex
+	dropTotal         uint64
+	drops             map[string]uint64
+	reportingInterval *durationpb.Duration
+}
+
+func (lrss *lrsServer) StreamLoadStats(stream lrspb.LoadReportingService_StreamLoadStatsServer) error {
+	req, err := stream.Recv()
+	if err != nil {
+		return err
+	}
+	if !proto.Equal(req, &lrspb.LoadStatsRequest{
+		Node: &basepb.Node{
+			Metadata: &structpb.Struct{
+				Fields: map[string]*structpb.Value{
+					grpcHostname: {
+						Kind: &structpb.Value_StringValue{StringValue: testService},
+					},
+				},
+			},
+		},
+	}) {
+		return status.Errorf(codes.FailedPrecondition, "unexpected req: %+v", req)
+	}
+	if err := stream.Send(&lrspb.LoadStatsResponse{
+		Clusters:              []string{testService},
+		LoadReportingInterval: lrss.reportingInterval,
+	}); err != nil {
+		return err
+	}
+
+	for {
+		req, err := stream.Recv()
+		if err != nil {
+			if err == io.EOF {
+				return nil
+			}
+			return err
+		}
+		stats := req.ClusterStats[0]
+		lrss.mu.Lock()
+		lrss.dropTotal += stats.TotalDroppedRequests
+		for _, d := range stats.DroppedRequests {
+			lrss.drops[d.Category] += d.DroppedCount
+		}
+		lrss.mu.Unlock()
+	}
+}
+
+func setupServer(t *testing.T, reportingInterval *durationpb.Duration) (addr string, lrss *lrsServer, cleanup func()) {
+	lis, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		t.Fatalf("listen failed due to: %v", err)
+	}
+	svr := grpc.NewServer()
+	lrss = &lrsServer{
+		drops:             make(map[string]uint64),
+		reportingInterval: reportingInterval,
+	}
+	lrspb.RegisterLoadReportingServiceServer(svr, lrss)
+	go svr.Serve(lis)
+	return lis.Addr().String(), lrss, func() {
+		svr.Stop()
+		lis.Close()
+	}
+}
+
+func Test_lrsStore_reportTo(t *testing.T) {
```

Comment:
Naming is not go style?

Reply:
I'm not sure. This is auto generated, and lint didn't complain. Maybe for tests lint is not as strict.